packages/wonder-blocks-floating/src/components/__tests__/floating.test.tsx

```diff
@@ -216,6 +216,180 @@ describe("Floating", () => {
+                    "First focusable element",
+                );
+
+                // Assert
+                await waitFor(() => {
+                    expect(firstFocusableElement).toHaveFocus();
+                });
+            });
+        });
+
+        describe("dismissEnabled", () => {
+            it("should not dismiss the floating element when dismissEnabled is false", async () => {
+                // Arrange
+                const onOpenChangeMock = jest.fn();
+
+                const {container} = render(
+                    <Floating
+                        content={
+                            <div>
+                                <button>First focusable element</button>
+                            </div>
+                        }
+                        open={true}
+                        dismissEnabled={false}
+                        onOpenChange={onOpenChangeMock}
+                    >
+                        <button>Trigger</button>
+                    </Floating>,
+                );
+
+                // Act
+                // Click outside the floating element
+                await userEvent.click(container);
+
+                // Assert
+                await waitFor(() => {
+                    expect(onOpenChangeMock).not.toHaveBeenCalled();
+                });
+            });
+
+            it("should dismiss the floating element when dismissEnabled is true", async () => {
+                // Arrange
+                const onOpenChangeMock = jest.fn();
+
+                const ControlledFloating = () => {
+                    const [open, setOpen] = React.useState(true);
+                    return (
+                        <Floating
+                            content={
+                                <div>
+                                    <button>First focusable element</button>
+                                </div>
+                            }
+                            dismissEnabled={true}
+                            open={open}
+                            onOpenChange={(open) => {
+                                setOpen(open);
+                                onOpenChangeMock(open);
+                            }}
+                        >
+                            <button>Trigger</button>
+                        </Floating>
+                    );
+                };
+
+                const {container} = render(<ControlledFloating />);
+
+                // Act
+                // Click outside the floating element
+                await userEvent.click(container);
+
+                // Assert
+                await waitFor(() => {
+                    // False because the floating element is dismissed
+                    expect(onOpenChangeMock).toHaveBeenCalledWith(false);
+                });
+            });
+
+            it("should dismiss the floating element when the escape key is pressed", async () => {
```

Comment:
Can we also add a test that pressing escape doesn't dismiss that floating element when `dismissEnabled` is false?

And also tests for the default behaviour when the `dismissEnabled` prop isn't provided!